src/clients/python/python_bindings.zig

```diff
@@ -0,0 +1,497 @@
+    .{ tb_client.tb_status_t, "Status" },
+};
+
+/// State machine specific mappings: in future, these should be pulled automatically from the state
+/// machine.
+const mappings_state_machine = .{
+    .{ tb.AccountFlags, "AccountFlags" },
+    .{ tb.TransferFlags, "TransferFlags" },
+    .{ tb.AccountFilterFlags, "AccountFilterFlags" },
+    .{ tb.QueryFilterFlags, "QueryFilterFlags" },
+    .{ tb.Account, "Account" },
+    .{ tb.Transfer, "Transfer" },
+    .{ tb.CreateAccountResult, "CreateAccountResult" },
+    .{ tb.CreateTransferResult, "CreateTransferResult" },
+    .{ tb.CreateAccountsResult, "CreateAccountsResult" },
+    .{ tb.CreateTransfersResult, "CreateTransfersResult" },
+    .{ tb.AccountFilter, "AccountFilter" },
+    .{ tb.AccountBalance, "AccountBalance" },
+    .{ tb.QueryFilter, "QueryFilter" },
+};
+
+const mappings_all = mappings_vsr ++ mappings_state_machine;
+
+const Buffer = struct {
+    inner: std.ArrayList(u8),
+
+    pub fn init(allocator: std.mem.Allocator) Buffer {
+        return .{
+            .inner = std.ArrayList(u8).init(allocator),
+        };
+    }
+
+    pub fn print(self: *Buffer, comptime format: []const u8, args: anytype) void {
+        self.inner.writer().print(format, args) catch unreachable;
+    }
+};
+
+fn mapping_name_from_type(mappings: anytype, Type: type) ?[]const u8 {
+    inline for (mappings) |mapping| {
+        const ZigType = mapping[0];
+        const name = mapping[1];
+
+        if (Type == ZigType) {
+            return name;
+        }
+    }
+    return null;
+}
+
+/// Resolves a Zig Type into a string representing the name of a corresponding Python ctype. This
+/// resolves both VSR and state machine specific mappings, as both are needed when interfacing via
+/// FFI.
+fn resolve_type_ctypes(comptime Type: type) []const u8 {
+    switch (@typeInfo(Type)) {
+        .Array => |info| {
+            return std.fmt.comptimePrint("{s} * {d}", .{
+                comptime resolve_type_ctypes(info.child),
+                info.len,
+            });
+        },
+        .Enum => |info| return resolve_type_ctypes(info.tag_type),
+        .Struct => return resolve_type_ctypes(std.meta.Int(.unsigned, @bitSizeOf(Type))),
+        .Bool => return "ctypes.c_bool",
+        .Int => |info| {
+            assert(info.signedness == .unsigned);
+            return switch (info.bits) {
+                8 => "ctypes.c_uint8",
+                16 => "ctypes.c_uint16",
+                32 => "ctypes.c_uint32",
+                64 => "ctypes.c_uint64",
+                128 => "c_uint128",
+                else => @compileError("invalid int type"),
+            };
+        },
+        .Optional => |info| switch (@typeInfo(info.child)) {
+            .Pointer => return resolve_type_ctypes(info.child),
+            else => @compileError("Unsupported optional type: " ++ @typeName(Type)),
+        },
+        .Pointer => |info| {
+            assert(info.size != .Slice);
```

Comment:
Let's state the invariant positively by switching on `info.size` and explicitly listing allowed variants. That way, if Zig adds a new variant to Size, we'll get notified instead of it being treated silently as non slice